superset/daos/tag.py

```diff
@@ -257,3 +266,90 @@ def get_tagged_objects_for_tags(
                 for obj in saved_queries
             )
         return results
+
+    @staticmethod
+    def user_favorite_tag(tag_id: int) -> None:
+        """
+        Marks a specific tag as a favorite for the current user.
+        This function will find the tag by the provided id,
+        create a new UserFavoriteTag object that represents
+        the user's preference, add that object to the database
+        session, and commit the session. It uses the currently
+        authenticated user from the global 'g' object.
+        Args:
+            tag_id: The id of the tag that is to be marked as
+                    favorite.
+        Raises:
+            Any exceptions raised by the find_by_id function,
+            the UserFavoriteTag constructor, or the database session's
+            add and commit methods will propagate up to the caller.
+        Returns:
+            None.
+        """
+        tag = TagDAO.find_by_id(tag_id)
+        user = g.user
+
+        if tag and user:
+            tag.users_favorited.append(user)
+            db.session.add(tag)
+            db.session.commit()
+
+    @staticmethod
+    def remove_user_favorite_tag(tag_id: int) -> None:
+        """
+        Removes a tag from the current user's favorite tags.
+
+        This function will find the tag by the provided id and remove the tag
+        from the user's list of favorite tags. It uses the currently authenticated
+        user from the global 'g' object.
+
+        Args:
+            tag_id: The id of the tag that is to be removed from the favorite tags.
+
+        Raises:
+            Any exceptions raised by the find_by_id function, the database session's
+            commit method will propagate up to the caller.
+
+        Returns:
+            None.
+        """
+        # Find the tag
+        tag = TagDAO.find_by_id(tag_id)
+        user = g.user
+
+        # Remove the tag from the user's favorites
+        if tag and user:
+            tag.users_favorited.remove(user)
+
+        # Commit to save the changes
+        db.session.commit()
+
+    @staticmethod
+    def favorited_ids(tags: list[Tag]) -> list[int]:
```

Comment:
naming: favorited_tag_ids

Reply:
I think it's better to just pass in tag IDs instead of the tag objects

Reply:
i want to follow the same convention i the other DAOs:
https://github.com/preset-io/superset/blob/84c183e0e6005738b0dad60a30824b17a087a752/superset/daos/chart.py#L73